bridge/client/app/_views/ktb-settings-view/ktb-integration-view/ktb-integration-view.component.ts

```diff
@@ -1,158 +1,106 @@
-      status = 1;
-    }
-    return status;
-  }
-
-  ngOnDestroy(): void {
-    this.unsubscribe$.next();
-    this.unsubscribe$.complete();
-  }
-
-  public setSelectedUniformRegistration(uniformRegistration: UniformRegistration): void {
-    if (this.selectedUniformRegistration !== uniformRegistration) {
-      this.lastSeen = this.dataService.getUniformDate(uniformRegistration.id);
-
-      uniformRegistration.unreadEventsCount = 0;
-      if (!this.uniformRegistrations.data.some((registration) => registration.unreadEventsCount !== 0)) {
-        this.dataService.setHasUnreadUniformRegistrationLogs(false);
-      }
-
-      this.selectedUniformRegistration = uniformRegistration;
-      this.selectedUniformRegistrationId$.next(this.selectedUniformRegistration.id);
+  ) {}
+
+  public setSelectedUniformRegistration(uniformRegistration: UniformRegistration, projectName: string): void {
+    const routeUrl = this.router.createUrlTree([
+      '/project',
+      projectName,
+      'settings',
+      'uniform',
+      'integrations',
+      uniformRegistration.id,
+    ]);
+    this.location.go(routeUrl.toString());
+    this.lastSeen = this.dataService.getUniformDate(uniformRegistration.id);
+    uniformRegistration.unreadEventsCount = 0;
+    const noUnreadLogs = this.uniformRegistrations.data.every((r) => r.unreadEventsCount === 0);
+    if (noUnreadLogs) {
+      this.dataService.setHasUnreadUniformRegistrationLogs(false);
     }
+    this.selectUniformRegistrationId$.next(uniformRegistration.id);
   }
 
   public sortData(sortEvent: DtSortEvent): void {
-    if (this.uniformRegistrations.data) {
-      const isAscending = sortEvent.direction === 'asc';
-      const data: UniformRegistration[] = this.uniformRegistrations.data.slice();
-
-      data.sort((a: UniformRegistration, b: UniformRegistration) => {
-        switch (sortEvent.active) {
-          case 'host':
-            return (
-              this.compare(a.metadata.hostname, b.metadata.hostname, isAscending) || this.compare(a.name, b.name, true)
-            );
-          case 'namespace':
-            return (
-              this.compare(
-                a.metadata.kubernetesmetadata.namespace,
-                b.metadata.kubernetesmetadata.namespace,
-                isAscending
-              ) || this.compare(a.name, b.name, true)
-            );
-          case 'location':
-            return (
-              this.compare(a.metadata.location, b.metadata.location, isAscending) || this.compare(a.name, b.name, true)
-            );
-          case 'name':
-          default:
-            return this.compare(a.name, b.name, isAscending);
-        }
-      });
-
-      this.uniformRegistrations.data = data;
-    } else {
-      this.uniformRegistrations.data = [];
-    }
+    this.uniformRegistrations.data = sortRegistrations(
+      this.uniformRegistrations.data,
+      sortEvent.active,
+      sortEvent.direction === 'asc'
+    );
   }
 
   public getOverlay(
```

Comment:
Could you maybe get rid of this unneeded function?
In the HTML you can just define
```HTML
[dtOverlay]="noRegistration"
[disabled]="toUniformRegistration(row).hasSubscriptions(params.projectName)"
```
`disabled` will deactivate the overlay.

Reply:
I tested it but it don't work :(

Reply:
Are you sure about that? I also tested it locally and it worked. Let's have a look at it together then :)